test/samples/meson.build

```diff
@@ -0,0 +1,63 @@
+# Create core video, audio, and subtitle streams to construct files out of.
+video = custom_target('video.mkv',
+    output: 'video.mkv',
+    command: [ffmpeg, '-y', '-f', 'lavfi', '-i', 'testsrc=duration=2:size=1280x720', '@OUTPUT@'],
+)
+
+audio = custom_target('audio.flac',
+    output: 'audio.flac',
+    command: [ffmpeg, '-y', '-f', 'lavfi', '-i', 'sine=frequency=1000:duration=2', '@OUTPUT@'],
+)
+
+sub = join_paths(source_root, 'test', 'samples', 'sub.srt')
+
+samples = {
+    'eng_default.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-f', 'srt', '-i', sub,
+         '-map', '0:0', '-map', '1:0', '-map', '2:0', '-c:v', 'copy',
+         '-c:a', 'copy', '-c:s', 'srt', '-metadata:s:s:0', 'language=eng',
+         '-disposition:s:0', 'default', '@OUTPUT@'],
+    'eng_default_forced.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-f', 'srt', '-i', sub,
+         '-map', '0:0', '-map', '1:0', '-map', '2:0', '-c:v', 'copy',
+         '-c:a', 'copy', '-c:s', 'srt', '-metadata:s:s:0', 'language=eng',
+         '-disposition:s:0', 'default+forced', '@OUTPUT@'],
+    'eng_forced_matching_audio.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-f', 'srt', '-i', sub,
+         '-map', '0:0', '-map', '1:0', '-map', '2:0', '-c:v', 'copy',
+         '-c:a', 'copy', '-c:s', 'srt', '-metadata:s:a:0', 'language=eng',
+         '-metadata:s:s:0', 'language=eng', '-disposition:s:0', 'forced', '@OUTPUT@'],
+    'eng_forced_no_matching_audio.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-f', 'srt', '-i', sub,
+         '-map', '0:0', '-map', '1:0', '-map', '2:0', '-c:v', 'copy',
+         '-c:a', 'copy', '-c:s', 'srt', '-metadata:s:s:0', 'language=eng',
+         '-disposition:s:0', 'forced', '@OUTPUT@'],
+    'eng_no_default.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-f', 'srt', '-i', sub,
+         '-map', '0:0', '-map', '1:0', '-map', '2:0', '-c:v', 'copy',
+         '-c:a', 'copy', '-c:s', 'srt', '-metadata:s:s:0', 'language=eng',
+         '@OUTPUT@'],
+    'multilang.mkv':
+        [ffmpeg, '-y', '-i', video, '-i', audio, '-i', audio, '-i', audio, '-i', audio,
+         '-f', 'srt', '-i', sub, '-i', sub, '-i', sub, '-i', sub, '-map', '0:0',
+         '-map', '1:0', '-map', '2:0', '-map', '3:0', '-map', '4:0', '-map', '5:0',
+         '-map', '6:0', '-map', '7:0', '-map', '8:0', '-metadata:s:a:0', 'language=eng',
+         '-metadata:s:a:1', 'language=jpn', '-metadata:s:a:2', 'language=ger', '-metadata:s:a:3',
+         'language=pol', '-metadata:s:s:0', 'language=eng', '-metadata:s:s:1', 'language=jpn',
+          '-metadata:s:s:2', 'language=ger', '-metadata:s:s:3', 'language=pol', '-disposition:s:0',
+          'default+forced', '-disposition:s:1', 'forced', '-disposition:s:2', 'default', '@OUTPUT@'],
+}
+
+sample_targets = []
+foreach name, cmd: samples
```

Comment:
Open question. Why not define `test()` for each of them? This will allow to see individual test results without looking up verbose output and remove loop form test itself, so it tests only one unit at the time.

Reply:
That's a good idea. With some minor tweaking it could work.